src/pages/invites/index.tsx

```diff
@@ -0,0 +1,236 @@
+// implement show all invites from local as list ordered from new to old
+import { FC, useState } from 'react';
+import {
+  deleteInviteLocally,
+  Invite,
+  makeInviteUrl,
+  updateInviteLocally,
+} from '@utils/invite';
+import Blockies from 'react-blockies';
+import Key from '@media/UI/invite/key.png';
+import NextImage from '@media/UI/invite/next.png';
+import BackImage from '@media/UI/invite/back.png';
+import EditImage from '@media/UI/invite/pencil.png';
+import SaveImage from '@media/UI/invite/check-mark.png';
+import DeleteImage from '@media/UI/invite/delete.png';
+import DollarImage from '@media/UI/invite/dollar-sign.png';
+import EmptyImage from '@media/UI/invite/empty.png';
+import copy from 'copy-to-clipboard';
+
+interface AllInvitesProps {
+  invites: Invite[];
+  updateInvites: () => void;
+  onTopUpInvite: (invite: Invite) => void;
+}
+
+export const ITEMS_PER_PAGE = 6;
+
+const Invites: FC<AllInvitesProps> = ({
+  invites,
+  updateInvites,
+  onTopUpInvite,
+}) => {
+  const [currentPage, setCurrentPage] = useState(0);
+  // keep track of the invite being edited
+  const [editingInvite, setEditingInvite] = useState<Invite>(null);
+  const [editingInviteName, setEditingInviteName] = useState('');
```

Comment:
The `editingInviteName` is a redundant data. The name can be used from `editingInvite` and updated there too.